File: 백준/Bronze/2851. 슈퍼 마리오/슈퍼 마리오.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int[] mushroomScores = new int[10];
        for (int i = 0; i < 10; i++) {
            mushroomScores[i] = Integer.parseInt(br.readLine().trim());
        }
        
        int currentSum = 0;
        int closestSum = 0;

        for (int i = 0; i < 10; i++) {
            currentSum += mushroomScores[i];
            if (currentSum >= 100) {
                if (Math.abs(100 - currentSum) < Math.abs(100 - closestSum)) {
                    closestSum = currentSum;
                } else if (Math.abs(100 - currentSum) == Math.abs(100 - closestSum)) {
                    closestSum = Math.max(currentSum, closestSum);
                }
                break;
            }
            closestSum = currentSum;
        }

        System.out.println(closestSum);
    }
}

File: 백준/Bronze/2851. 슈퍼 마리오/슈퍼 마리오.py
# 버섯 점수를 입력받아 리스트로 저장
mushroom_scores = []
for _ in range(10):
    mushroom_scores.append(int(input().strip()))

# 초기화
current_sum = 0
closest_sum = 0

# 점수를 누적하며 계산
for score in mushroom_scores:
    current_sum += score
    if current_sum >= 100:
        # 100을 초과했을 때 이전의 누적합과 현재 누적합 중 100에 가까운 값을 선택
        if abs(100 - current_sum) < abs(100 - closest_sum):
            closest_sum = current_sum
        elif abs(100 - current_sum) == abs(100 - closest_sum):
            closest_sum = max(current_sum, closest_sum)
        break
    closest_sum = current_sum

print(closest_sum)
